mithril-common/src/chain_observer/pallas_observer.rs

```diff
@@ -566,4 +627,28 @@ mod tests {
 
         assert_eq!(expected_stake_distribution, computed_stake_distribution);
     }
+
+    #[tokio::test]
+    async fn get_current_kes_period() {
+        let socket_path = create_temp_dir("get_current_kes_period").join("node.socket");
+        let server = setup_server(socket_path.clone()).await;
+        let client = tokio::spawn(async move {
+            let observer =
+                super::PallasChainObserver::new(socket_path.as_path(), CardanoNetwork::TestNet(10));
+
+            let keypair = ColdKeyGenerator::create_deterministic_keypair([0u8; 32]);
+            let mut dummy_key_buffer = [0u8; Sum6Kes::SIZE + 4];
+            let mut dummy_seed = [0u8; 32];
+            let (_, kes_verification_key) = Sum6Kes::keygen(&mut dummy_key_buffer, &mut dummy_seed);
+            let operational_certificate = OpCert::new(kes_verification_key, 0, 0, keypair);
+            observer
+                .get_current_kes_period(&operational_certificate)
+                .await
+                .unwrap()
+        });
+
+        let (_, client_res) = tokio::join!(server, client);
+        let kes_period = client_res.unwrap().unwrap();
+        assert_eq!(407, kes_period);
```

Comment:
We should probably test that the result is corresponding to the correct computation of an integer division with multiple cases: the computation is critical to the key registration and could block the whole protocol if done incorrectly.